File: example-js/puzzle.js
import { createAocPuzzle } from "./aoc.js";

const testAnswers = {
  part1: 1379,
  part2: 56154
};

function parseInput(input) {
  return input.map(line => +line);
}

function runPart1(input) {
  return input.reduce((prev, cur) => prev + cur);
}

function runPart2(input) {
  return input.reduce((prev, cur) => prev * cur);
}

export function createPuzzle() {
  return createAocPuzzle({ parseInput, runPart1, runPart2, testAnswers });
}

File: example-py/puzzle.py
from functools import reduce
from aoc import AocPuzzle


TEST_ANSWERS = {
    "part1": 1379,
    "part2": 56154
}


class Puzzle(AocPuzzle):
    def __init__(self):
        super().__init__(TEST_ANSWERS)


    def parse_input(self, lines):
        return [int(line) for line in lines]


    def run_part1(self):
        return sum(self._input)


    def run_part2(self):
        return reduce(lambda prev, cur: prev * cur, self._input)
